File: reference/two-pointer/844-Backspace-String-Compare.go
func backspaceCompare(s string, t string) bool {
    stackStringCompare := func(str string) []rune {
        stack := []rune{}
        for _, char := range str {
            if char != '#' {
                stack = append(stack, char)
            } else if len(stack) > 0 {
                stack = stack[:len(stack) - 1]
            }
        }
        return stack
    }

    return string(stackStringCompare(s)) == string(stackStringCompare(t))
}

File: reference/two-pointer/844-Backspace-String-Compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        
        def stack_string_compare(string):
            stack = []
            for char in string:
                if char != "#":
                    stack.append(char)
                elif stack:
                    stack.pop()
            return stack
        
        return stack_string_compare(s) == stack_string_compare(t)